src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1097,8 +1091,8 @@ TxQ::apply(
             // Create the test view from the current view.
             multiTxn.emplace(view, flags);
 
-            auto const sleBump = multiTxn->applyView.peek(accountKey);
-            if (!sleBump)
+            auto acctRootBump = multiTxn->applyView.peek(accountKey);
```

Comment:
ok, thanks for the clarification. 

In the existing code, `auto const sleBump` would prevent us from updating the control block of the shared_ptr, since it is marked as const, am I right? But that goes against the functionality of a shared pointer (incrementing and decrementing the number of references to the SLE object)

Reply:
I'm pretty sure that with `auto const sleBump` the control block is still mutable.  It's the local representation of the `std::shared_ptr` that is `const`.  So you can't, for example, assign to it.  All you can do to the local representation is destroy it, which would have the effect of decrementing the reference count in the control block.